homeassistant/components/sensor/entur_public_transport.py

```diff
@@ -0,0 +1,330 @@
+      }
+      destinationDisplay {
+        frontText
+      }
+      serviceJourney {
+        journeyPattern {
+          line {
+            id
+            name
+            transportMode
+          }
+        }
+      }
+    }
+  }
+"""
+_GRAPHQL_QUAY_TEMPLATE = """
+  quays(ids:[$quays]) {
+    id
+    name
+    estimatedCalls(
+        startTime: \"$time\",
+        timeRange: 72100,
+        numberOfDepartures: 2) {
+      realtime
+      aimedArrivalTime
+      aimedDepartureTime
+      expectedArrivalTime
+      expectedDepartureTime
+      requestStop
+      notices {
+        text
+      }
+      destinationDisplay {
+        frontText
+      }
+      serviceJourney {
+        journeyPattern {
+          line {
+            id
+            name
+            transportMode
+          }
+        }
+      }
+    }
+  }
+"""
+
+ATTR_STOP_ID = 'Stop ID'
+
+ATTR_ROUTE = 'Route'
+ATTR_EXPECTED_IN = 'Due in'
+ATTR_EXPECTED_AT = 'Due at'
+ATTR_DELAY = 'Delay'
+ATTR_REALTIME = 'Real-time'
+
+ATTR_NEXT_UP_ROUTE = 'Next departure route'
+ATTR_NEXT_UP_IN = 'Next departure in'
+ATTR_NEXT_UP_AT = 'Next departure at'
+ATTR_NEXT_UP_DELAY = 'Next departure delay'
+ATTR_NEXT_UP_REALTIME = 'Next departure is real-time'
+
+CONF_ATTRIBUTION = "Data provided by entur.org under NLOD."
+CONF_STOP_IDS = 'stop_ids'
+
+ICON = 'mdi:bus'
+
+SCAN_INTERVAL = timedelta(minutes=1)
+TIME_STR_FORMAT = '%H:%M'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STOP_IDS): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+def due_in_minutes(timestamp: str) -> str:
+    """Get the time in minutes from a timestamp.
+
+    The timestamp should be in the format
+    year-month-yearThour:minute:second+timezone
+    """
+    if timestamp is None:
+        return 'Unknown'
+    diff = datetime.strptime(
+        timestamp, "%Y-%m-%dT%H:%M:%S%z") - dt_util.now()
+
+    return str(int(diff.total_seconds() / 60))
+
+
+def time_diff_in_minutes(timestamp1: str, timestamp2: str) -> str:
+    """Get the time in minutes from a timestamp.
+
+    The timestamp should be in the format
+    year-month-yearThour:minute:second+timezone
+    """
+    if timestamp1 is None:
+        return 'Unknown'
+    if timestamp2 is None:
+        return 'Unknown'
+
+    time1 = datetime.strptime(timestamp1, "%Y-%m-%dT%H:%M:%S%z")
+    time2 = datetime.strptime(timestamp2, "%Y-%m-%dT%H:%M:%S%z")
+    diff = time1 - time2
+
+    return str(int(diff.total_seconds() / 60))
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Dublin public transport sensor."""
+    stop_ids = config.get(CONF_STOP_IDS)
+
+    stops = [s for s in stop_ids if "StopPlace" in s]
+    quays = [s for s in stop_ids if "Quay" in s]
+
+    data = PublicTransportData(stops, quays)
+    data.update()
+    entities = []
+    for item in stop_ids:
+        entities.append(EnturPublicTransportSensor(data, item))
+
+    add_entities(entities, True)
+
+
+class PublicTransportData:
```

Comment:
This code needs to be moved to a module which is [published on PyPI](https://developers.home-assistant.io/docs/en/development_checklist.html) and imported.

Reply:
Ok, will try to learn PyPI as well and update the PR 👍

Reply:
I hope everything is in order now. Api calls are now done through dependecy from PyPi.